### DISCUSSION TITLE
Binding search in transformer?

### DISCUSSION THREAD

**Discussion by shrinktofit** [swc-project/swc#2017] [Q&A]
I'm new to swc. I'm attempting to implement SystemJS transformer.

How to do binding search in tranformer code like babel's scope? For example search all bindings to an indent to find out it's declaration AST.

I've looked into `Scope` in `ecmascript\transforms\module\src\util.rs`', but it seems not what I need.

**Comment by kdy1** [comment]
Do you want to find all used identifiers? If so, you can use `Visit` (https://docs.rs/swc_visit/0.2.5/swc_visit/) to collect them. You can store variables from nested scope into a single vector or map, thanks to `resolver`. See https://docs.rs/swc_ecma_transforms_base/0.26.0/swc_ecma_transforms_base/resolver/fn.resolver_with_mark.html

**Reply by kdy1** [reply]
Update: theres' https://docs.rs/swc_ecma_utils/0.101.3/swc_ecma_utils/fn.collect_decls.html

**Comment by johnhere6754** [comment]
Therefore, the search for a promising molecule that may bind to a target protein is a challenging task for chemists. Building on [Colt Python for sale](https://hodgdonretailstore.com/product/new-colt-python-for-sale/?fbclid=IwAR0yJl7Iu3z4NyWxAB9KiRbMxTdTRta3-aQU5Dm-klw-bToJMcf6fHcfTyM) legacy, the legendary double-action revolver returns in stainless steel in 4.25” and 6” barrel lengths.
